### PR TITLE
refactor(ci): replace grep/sed bundle parsing with yq

### PR DESCRIPTION
**Pull request by sonupreetam**
## Summary

- Replaces fragile `grep -E '^\s*-\s+'` + `sed` bundle layer extraction with proper `yq` queries
- Guards all `yq` expressions against missing/empty/null `layers` using yq's alternative operator (`//`), preventing non-zero exits under `pipefail`
- Adds automated test infrastructure (`scripts/test-bundle-parsing.sh`, `Makefile`, CI job) covering edge cases and real bundles
- Affects two locations in `publish-policy-oci.yml`: the discover job (root file extraction) and the content hashing step
- Affects one location in `cue-validate.yml`: the bundle layer existence check
- `yq` is pre-installed on `ubuntu-latest` runners — no additional setup needed

## Motivation

Raised in [PR #41 review](https://github.com/complytime/complytime-policies/pull/41#discussion_r3319064753) (LOW priority). The `grep` pattern works for the current simple bundle format but would silently break if bundles ever include comments, multi-line values, or non-standard whitespace.

### `publish-policy-oci.yml`
- **Discover job**: `grep | tail | sed` → `yq -r '(.layers // []) | select(length > 0) | .[-1]'`
- **Hash step**: `grep | sed` → `yq -r '(.layers // [])[]'`

### `cue-validate.yml`
- **Layer existence check**: `grep | sed` → `yq -r '(.layers // [])[]'`
- **New CI job**: `test-bundle-parsing` runs `make test-bundle-parsing` on PRs
- **Path triggers**: expanded to include `.github/workflows/**` and `scripts/**`

### New files
- `scripts/test-bundle-parsing.sh` — 16 tests (synthetic edge cases + real bundles) under `pipefail`
- `Makefile` — `test-bundle-parsing` and `test` targets

## Pipefail safety

The raw `yq '.layers[-1]'` expression exits non-zero when `layers` is missing, null, or empty:

| Scenario | `.layers[-1]` (old) | `(.layers // []) \| select(length > 0) \| .[-1]` (fixed) |
|---|---|---|
| No `layers` key | exit 1 — crashes | exit 0, empty string |
| `layers: []` | exit 1 — crashes | exit 0, empty string |
| `layers:` (null) | exit 1 — crashes | exit 0, empty string |
| Valid layers | exit 0, correct | exit 0, correct |

## Test plan

- [x] Verify `yq -r '(.layers // []) | select(length > 0) | .[-1]' bundles/ampel-branch-protection.yaml` returns the correct root file
- [x] Verify `yq -r '(.layers // [])[]' bundles/ampel-branch-protection.yaml` returns all layer paths
- [x] All 3 expressions return empty/exit 0 for missing, null, and empty `layers` under `pipefail`
- [x] `make test-bundle-parsing` passes locally (16/16 tests)
- [ ] CI passes on the CUE validation workflow (includes new test-bundle-parsing job)

### CODE REVIEW

File: .github/workflows/publish-policy-oci.yml [hunk 1]
```diff
@@ -66,7 +66,7 @@ jobs:
             [ -f "${bundle_file}" ] || continue
             bundle_name="$(basename "${bundle_file}" .yaml)"
             # The root policy file is the last layer entry (type: Policy)
-            root_file="$(grep -E '^\s*-\s+' "${bundle_file}" | tail -1 | sed 's/^[[:space:]]*-[[:space:]]*//')"
+            root_file="$(yq '.layers[-1]' "${bundle_file}")"
```
Comment (trevor-vaughan) [inline .github/workflows/publish-policy-oci.yml]: This will crash the job if layers is not found due to `pipefail` so the `skipping` message will never be reached and the job will error-out.

**Review by trevor-vaughan — CHANGES_REQUESTED** [review]
We should run through a set of tests with `yq` to make sure that the jobs aren't going to die incorrectly based on `pipefail` being (correctly) set with artifacts that do not have a `layers` object.

File: .github/workflows/publish-policy-oci.yml (hunk 1 above)
Comment (sonupreetam) [reply .github/workflows/publish-policy-oci.yml]: @trevor-vaughan Thank you for catching this — you were right.

**Review by trevor-vaughan — APPROVED** [review]

**Review by hbraswelrh — APPROVED** [review]
LGTM.
